src/main/frontend/pipeline-console-view/pipeline-console/main/ConsoleLogCard.tsx

```diff
@@ -0,0 +1,239 @@
+  handleMoreConsoleClick: (nodeId: string, startByte: number) => void;
+  // Id of the element whose scroll bar we wish to use.
+  scrollParentId: string;
+};
+
+export class ConsoleLogCard extends React.Component<ConsoleLogCardProps> {
+  constructor(props: ConsoleLogCardProps) {
+    super(props);
+    this.handleStepToggle = this.handleStepToggle.bind(this);
+  }
+
+  handleStepToggle(event: React.MouseEvent<HTMLElement>) {
+    this.props.handleStepToggle(event, String(this.props.step.id));
+  }
+
+  getRowHeight(index: number) {
+    return 20;
+  }
+
+  getTrucatedLogWarning() {
+    if (this.props.step.consoleLines && this.props.step.consoleStartByte != 0) {
+      return (
+        <Grid container>
+          <Grid item xs={6} sm className="show-more-console">
+            <Typography align="right" className="step-header">
+              {`Missing ${this.prettySizeString(
+                this.props.step.consoleStartByte
+              )} of logs.`}
+            </Typography>
+          </Grid>
+          <Grid item xs={6} sm className="show-more-console">
+            <Button
+              variant="text"
+              sx={{ padding: "0px", textTransform: "none" }}
+              onClick={() => {
+                let startByte =
+                  this.props.step.consoleStartByte - LOG_FETCH_SIZE;
+                console.debug(
+                  `startByte '${this.props.step.consoleStartByte}' -> '${startByte}'`
+                );
+                if (startByte < 0) {
+                  startByte = 0;
+                }
+                this.props.handleMoreConsoleClick(
+                  String(this.props.step.id),
+                  startByte
+                );
+              }}
+            >
+              Show more logs
+            </Button>
+          </Grid>
+        </Grid>
+      );
+    } else {
+      <div></div>;
+    }
+  }
+
+  prettySizeString(size: number) {
+    let kib = 1024;
+    let mib = 1024 * 1024;
+    let gib = 1024 * 1024 * 1024;
+    if (size < kib) {
+      return `${size}b`;
+    } else if (size < mib) {
+      return `${(size / kib).toFixed(2)}KiB`;
+    } else if (size < gib) {
+      return `${(size / mib).toFixed(2)}MiB`;
+    }
+    return `${(size / gib).toFixed(2)}GiB`;
+  }
+
+  renderSimpleConsoleLines() {
+    if (this.props.step.consoleLines) {
+      return this.props.step.consoleLines.map(
+        (content: string, index: number) => {
+          return (
+            <ConsoleLine
+              lineNumber={String(index)}
+              content={content}
+              stepId={String(this.props.step.id)}
+              startByte={this.props.step.consoleStartByte}
+            />
+          );
+        }
+      );
+    } else {
+      return <div></div>;
+    }
+  }
+
+  renderConsoleLine(index: number) {
+    if (this.props.step.consoleLines) {
+      return (
+        <ConsoleLine
+          lineNumber={String(index)}
```

Comment:
unsure if this is enough but what do you think about line numbers starting at 1? it seems a bit weird to me right now that they start at 0
```suggestion
          lineNumber={String(index + 1)}
```